zrml/court/src/migrations.rs

```diff
@@ -14,3 +14,220 @@
+use frame_support::{
+    log,
+    pallet_prelude::{StorageVersion, ValueQuery, Weight},
+    traits::{Get, OnRuntimeUpgrade},
+    BoundedVec,
+};
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::CheckedDiv, SaturatedConversion};
+
+#[derive(Decode, Encode, MaxEncodedLen, TypeInfo, Clone, Debug, PartialEq, Eq)]
+pub struct OldCourtPoolItem<AccountId, Balance, BlockNumber> {
+    pub stake: Balance,
+    pub court_participant: AccountId,
+    pub consumed_stake: Balance,
+    pub joined_at: BlockNumber,
+}
+
+type OldCourtPoolItemOf<T> = OldCourtPoolItem<AccountIdOf<T>, BalanceOf<T>, BlockNumberOf<T>>;
+type OldCourtPoolOf<T> = BoundedVec<OldCourtPoolItemOf<T>, <T as Config>::MaxCourtParticipants>;
+
+#[frame_support::storage_alias]
+pub(crate) type CourtPool<T: Config> = StorageValue<Court<T>, OldCourtPoolOf<T>, ValueQuery>;
+
+const COURT_REQUIRED_STORAGE_VERSION: u16 = 2;
+const COURT_NEXT_STORAGE_VERSION: u16 = 3;
+
+pub struct MigrateCourtPoolItems<T>(PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for MigrateCourtPoolItems<T>
+where
+    T: Config,
+{
+    fn on_runtime_upgrade() -> Weight {
+        let mut total_weight = T::DbWeight::get().reads(1);
+        let market_commons_version = StorageVersion::get::<Court<T>>();
+        if market_commons_version != COURT_REQUIRED_STORAGE_VERSION {
+            log::info!(
+                "MigrateCourtPoolItems: court storage version is {:?}, but {:?} is required",
+                market_commons_version,
+                COURT_REQUIRED_STORAGE_VERSION,
+            );
+            return total_weight;
+        }
+        log::info!("MigrateCourtPoolItems: Starting...");
+
+        let res = crate::CourtPool::<T>::translate::<OldCourtPoolOf<T>, _>(|old_pool_opt| {
+            old_pool_opt.map(|mut old_pool| {
+                <CourtPoolOf<T>>::truncate_from(
+                    old_pool
+                        .iter_mut()
+                        .map(|old_pool_item: &mut OldCourtPoolItemOf<T>| CourtPoolItemOf::<T> {
+                            stake: old_pool_item.stake,
+                            court_participant: old_pool_item.court_participant.clone(),
+                            consumed_stake: old_pool_item.consumed_stake,
+                            joined_at: old_pool_item.joined_at,
+                            uneligible_index: old_pool_item
+                                .joined_at
+                                .checked_div(&T::InflationPeriod::get())
+                                // because inflation period is not zero checked_div is safe
+                                // however, if not use the current period index
+                                // 5_184_000 is the block of the inflation reward distribution
+                                // 5_184_000 / 216_000 (blocks per 30 day inflation period) = 24
+                                .unwrap_or(24u64.saturated_into::<BlockNumberOf<T>>()),
```

Comment:
I think we should error in favor of the user. Setting it to the current period index would result in nobody getting any inflation payouts.

```suggestion
                                .checked_div(&T::InflationPeriod::get())
                                // because inflation period is not zero checked_div is safe
                                // however, if not use the current period index
                                // 5_184_000 is the block of the inflation reward distribution
                                // 5_184_000 / 216_000 (blocks per 30 day inflation period) = 24
                                .unwrap_or(0u64.saturated_into::<BlockNumberOf<T>>()),
```

Reply:
Besides that, this would also work for any chain using that pallet.

Reply:
9067fbc17dffd7bd1952b3d76e689e466b5ab515